packages/flutter/lib/src/material/material_state.dart

```diff
@@ -282,6 +288,123 @@ class _EnabledAndDisabledMouseCursor extends MaterialStateMouseCursor {
   String get debugDescription => 'MaterialStateMouseCursor($name)';
 }
 
+/// Defines a [BorderSide] whose value depends on a set of [MaterialState]s
```

Comment:
To make this class and MaterialStateOutlinedBorder easier to use, it would help to mimic the rest of MaterialStateColor. Apps and examples could write MaterialStateBorderSide.resolveWith(myCallback), which can take a bit of the sting out of using a MaterialStateProperty.

Reply:
Follow up issue for this: https://github.com/flutter/flutter/issues/69203